cpp/Embeddings.cc

```diff
@@ -0,0 +1,56 @@
+#include <cstdlib>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+
+#include "Embeddings.hh"
+#include "finalfusion.h"
+
+class Embeddings::EmbeddingsImpl {
+  friend Embeddings;
```

Comment:
Avoid `friend` unless absolutely necessary. It increases coupling. PIMP can be implemented using simple composition.